code/modules/emoji/emoji_parse.dm

```diff
@@ -19,7 +19,7 @@ var/list/emojis
 			if(search)
 				emoji = lowertext(copytext(text, pos+1, search))
 				if(emoji in emojis)
-					parsed += " <img class=icon src=\ref['icons/emoji.dmi'] iconstate='[emoji]'>"
+					parsed += bicon(icon('icons/emoji.dmi', emoji))
```

Comment:
In the end i used js and base64 css file for client-side emoticons, but examining of animated icons is sill an issue.